cmd/buildtools/utils.go

```diff
@@ -11,12 +12,90 @@ import (
 	"os"
 	"path/filepath"
 	"regexp"
+	"sort"
+	"strconv"
 	"strings"
 
 	"github.com/google/go-github/v62/github"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	wolfiAPKIndexURL = "https://packages.wolfi.dev/os/x86_64/APKINDEX.tar.gz"
+)
+
+func GetWolfiAPKIndex() ([]byte, error) {
+	tmpdir, err := os.MkdirTemp("", "wolfi-apk-index")
+	if err != nil {
+		return nil, fmt.Errorf("create temp dir: %w", err)
+	}
+	defer os.RemoveAll(tmpdir)
+	if err := DownloadFile(wolfiAPKIndexURL, filepath.Join(tmpdir, "APKINDEX.tar.gz")); err != nil {
+		return nil, fmt.Errorf("download APKINDEX.tar.gz: %w", err)
+	}
+	if err := ExtractTGZArchive(filepath.Join(tmpdir, "APKINDEX.tar.gz"), tmpdir); err != nil {
+		return nil, fmt.Errorf("extract APKINDEX.tar.gz: %w", err)
+	}
+	contents, err := os.ReadFile(filepath.Join(tmpdir, "APKINDEX"))
+	if err != nil {
+		return nil, fmt.Errorf("read APKINDEX: %w", err)
+	}
+	return contents, nil
+}
+
+func GetWolfiPackageVersion(wolfiAPKIndex []byte, pkgName string, pinnedVersion string) (string, error) {
+	// example APKINDEX content:
+	// P:calico-node
+	// V:3.26.1-r1
+	// ...
+	//
+	// P:calico-node
+	// V:3.26.1-r10
+	// ...
+	//
+	// P:calico-node
+	// V:3.26.1-r9
+	// ...
+
+	var revisions []int
+	scanner := bufio.NewScanner(bytes.NewReader(wolfiAPKIndex))
+	for scanner.Scan() {
+		line := scanner.Text()
+		// filter by package name
+		if !strings.HasPrefix(line, "P:"+pkgName) {
+			continue
+		}
+		scanner.Scan()
+		line = scanner.Text()
+		// filter by package version
+		if !strings.HasPrefix(line, "V:"+pinnedVersion+"-r") {
+			continue
+		}
+		// find the revision number
+		parts := strings.Split(line, "-r")
+		if len(parts) != 2 {
+			return "", fmt.Errorf("incorrect number of parts in APKINDEX line: %s", line)
+		}
+		revision, err := strconv.Atoi(parts[1])
+		if err != nil {
+			return "", fmt.Errorf("parse revision: %w", err)
+		}
+		revisions = append(revisions, revision)
+	}
+
+	if len(revisions) == 0 {
+		return "", fmt.Errorf("package %q not found", pkgName)
+	}
+
+	// get the latest revision
+
+	sort.Slice(revisions, func(i, j int) bool {
```

Comment:
you should probably handle scanner.Err since there is more than one revision and you could get a partial list if it were to fail